### PR TITLE
Add MSF tool md5_lookup.rb for MD5 searching

### PR DESCRIPTION
**Pull request by wchen-r7**
This is a msf tool based on @hasherezade's auxiliary module md5_lookup.rb (see #4399). I decided to port it as a tool because I find this more versatile. In order to use it, you don't have to load almost 2800 Metasploit modules. There may be some benefits of having this as a module or a plugin (personally I think plugin would be better) if it grabs hashes from the workspace, but db.rb currently does not support MD5 type so I did not pursuit that route. Anyways...
## Testing
- [x] First, make sure Travis is green
- [x] Before testing this tool, please save the following as a file. Let's save it as **/tmp/md5.txt**

```
dbd014125a4bad51db85f27279f1040a
098f6bcd4621d373cade4e832627b4f6
b5c0b187fe309af0f4d35982fd961d7e
a4757d7419ff3b48e92e90596f0e7548
63a9f0ea7bb98050796b649e85481845
21232f297a57a5a743894a0e4a801fc3
202cb962ac59075b964b07152d234b70
3e194b5730ed8b868224b80ac8d7eed9
5f4dcc3b5aa765d61d8327deb882cf99
f25a2fc72690b780b2a14e140ef6a9e0
```
- [x] Do `tools/md5_lookup.rb -h`, and make sure you see the help menu.
- [x] Do `tools/md5_lookup.rb -i /tmp/md5.txt`, it should "crack" all 10 hashes
- [x] After the script is done running, you should also see a **md5_results.txt** saved under the same directory
- [x] Make sure that **md5_results.txt** contains the original hashes, and the cracked ones
- [x] Now, let's do another search with these args: `tools/md5_lookup.rb -i /tmp/md5.txt -d i337 -o /tmp/results.txt`
- [x] All 10 hashes should be cracked/found again. But this time, on the screen you should also see all of them coming from **i337.net**.
- [x] Make sure the found results are saved as **/tmp/results.txt**
## Demo

screen shot 2015-01-17 at 2 07 12 am

### CODE REVIEW

**Comment by mubix** [comment]
Please add a warning to this script that it is looking up this information from a bunch of third parties. This could put pentesters in very hot water if they look up `c15663334624940d6a2533cf6303c2a9` (company1234) and those MD5 cracking sites see that its coming from within company.com's IP range. It's a nit pick but we "metasploit peeps" probably don't want to facilitate it.

**Comment by wchen-r7** [comment]
How do you word that properly from a legal standpoint?

@todb-r7 ?

**Comment by todb-r7** [comment]
@wchen-r7 I see what @mubix is saying, just need to warn the user that this tool makes external connections. Passing md5's of internal passwords to external entities, in the clear (SSL seems unavailable), is a risky thing. I'd say spell it out in the help text at a minimum.

Just eyeballing design, since you require msf/core, it'd be nice to save the results in a workspace-defined loot location (optionally). Not a blocker, but a nice to have for integration. If you were feeling particularly integration-happy, save the results in an importable format to populate a Metasploit creds table, or save directly there, too (again, optionally).

**Comment by todb-r7** [comment]
It's cool, though. I like it. Very handy.

**Comment by todb-r7** [comment]
Howabout an Are You Sure? Like so:

```
$ tools/md5_lookup.rb -i /tmp/md5.txt
[!] Warning, this tool will pass /tmp/md5.txt in the clear to external websites. Are you sure? [Yy]: y
[*] Great, let's get cracking!
...
```

And avoidable with `-y`:

```
$ tools/md5_lookup.rb -y -i /tmp/md5.txt
[*] Great, let's get cracking!
...
```

**Comment by wchen-r7** [comment]
Can do.

**Comment by wchen-r7** [comment]
Done.

Now when you run the tool for the very first time, you will see this:

```
$ ./md5_lookup.rb -i /tmp/md5.txt
[*] WARNING: This tool will look up your MD5 hashes by submitting them
[*] in the clear (HTTP) to third party websites.
[*] 
[*] The services from these third party websites are not within control
[*] of Metasploit (or Rapid7), and we shall not be liable for any loss or
[*] damage of whatever nature (direct, indirect, consequential, or other)
[*] as a result of your use of these services.
[*] 
[*] Enter 'Y' to acknowledge:
```

After you ack, you will not see it again.

**Comment by todb-r7** [comment]
For your approval, @wchen-r7 : wchen-r7/metasploit-framework#24

**Comment by wchen-r7** [comment]
Done.

**Comment by jvazquez-r7** [comment]
Verification:

```
$ tools/md5_lookup.rb -h
Usage: tools/md5_lookup.rb [options]

Specific options:
    -i, --input <file>               The file that contains all the MD5 hashes (one line per hash)
    -d, --databases <names>          (Optional) Select databases: all, authsecu, i337, md5_my_addr, md5_net, md5crack, md5cracker, md5decryption, md5online, md5pass, netmd5crack, tmto (Default=all)
    -o, --out <filepath>             (Optional) Save the results to a file (Default=md5_results.txt)
    -h, --help                       Show this message
$ tools/md5_lookup.rb -i /tmp/md5.txt
[*] WARNING: This tool will look up your MD5 hashes by submitting them
[*] in the clear (HTTP) to third party websites. This can expose
[*] sensitive data to unknown and untrusted entities.
[*] Enter 'Y' to acknowledge: Y
[*] Found: dbd014125a4bad51db85f27279f1040a = md5cracker.org (from i337.net)
[*] Found: 098f6bcd4621d373cade4e832627b4f6 = test (from authsecu)
[*] Found: b5c0b187fe309af0f4d35982fd961d7e = love (from i337.net)
[*] Found: a4757d7419ff3b48e92e90596f0e7548 = god (from i337.net)
[*] Found: 63a9f0ea7bb98050796b649e85481845 = root (from authsecu)
[*] Found: 21232f297a57a5a743894a0e4a801fc3 = admin (from authsecu)
[*] Found: 202cb962ac59075b964b07152d234b70 = 123 (from authsecu)
[*] Found: 3e194b5730ed8b868224b80ac8d7eed9 = texas (from md5.my-addr.com)
[*] Found: 5f4dcc3b5aa765d61d8327deb882cf99 = password (from authsecu)
[*] Found: f25a2fc72690b780b2a14e140ef6a9e0 = iloveyou (from i337.net)
$ cat md5_results.txt
dbd014125a4bad51db85f27279f1040a = md5cracker.org
098f6bcd4621d373cade4e832627b4f6 = test
b5c0b187fe309af0f4d35982fd961d7e = love
a4757d7419ff3b48e92e90596f0e7548 = god
63a9f0ea7bb98050796b649e85481845 = root
21232f297a57a5a743894a0e4a801fc3 = admin
202cb962ac59075b964b07152d234b70 = 123
3e194b5730ed8b868224b80ac8d7eed9 = texas
5f4dcc3b5aa765d61d8327deb882cf99 = password
f25a2fc72690b780b2a14e140ef6a9e0 = iloveyou
$ tools/md5_lookup.rb -i /tmp/md5.txt -d i337 -o /tmp/results.txt
[*] Found: dbd014125a4bad51db85f27279f1040a = md5cracker.org (from i337.net)
[*] Found: 098f6bcd4621d373cade4e832627b4f6 = test (from i337.net)
[*] Found: b5c0b187fe309af0f4d35982fd961d7e = love (from i337.net)
[*] Found: a4757d7419ff3b48e92e90596f0e7548 = god (from i337.net)
[*] Found: 63a9f0ea7bb98050796b649e85481845 = root (from i337.net)
[*] Found: 21232f297a57a5a743894a0e4a801fc3 = admin (from i337.net)
[*] Found: 202cb962ac59075b964b07152d234b70 = 123 (from i337.net)
[*] Found: 5f4dcc3b5aa765d61d8327deb882cf99 = password (from i337.net)
[*] Found: f25a2fc72690b780b2a14e140ef6a9e0 = iloveyou (from i337.net)
$ cat /tmp/results.txt
dbd014125a4bad51db85f27279f1040a = md5cracker.org
098f6bcd4621d373cade4e832627b4f6 = test
b5c0b187fe309af0f4d35982fd961d7e = love
a4757d7419ff3b48e92e90596f0e7548 = god
63a9f0ea7bb98050796b649e85481845 = root
21232f297a57a5a743894a0e4a801fc3 = admin
202cb962ac59075b964b07152d234b70 = 123
5f4dcc3b5aa765d61d8327deb882cf99 = password
f25a2fc72690b780b2a14e140ef6a9e0 = iloveyou
```

**Comment by jvazquez-r7** [comment]
Reviewing a little bit source, and landing!

File: tools/md5_lookup.rb [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+    attr_accessor :target_uri
+
+    # @!attribute ssl
+    #  @return [FalseClass] False because doesn't look like md5cracker.org supports HTTPS
+    attr_accessor :ssl
+
+    def initialize(opts={})
+      # The user should not be able to modify these settings, otherwise
+      # the we can't guarantee results.
+      self.rhost      = 'md5cracker.org'
```
Comment (jvazquez-r7) [inline tools/md5_lookup.rb]: Maybe should be constants? Not an stopper to me, just curiosity! I won't hold up landing!

**Comment by jvazquez-r7** [comment]
Landed! super useful tool!, thanks @wchen-r7 !

**Comment by wchen-r7** [comment]
Oh nice, thanks!

**Pull request merged**